### DISCUSSION TITLE
part-ii/2020-87/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#111] [Question comments]
Part II 2020 Logic and Set Theory: Paper 2, Section II, H 

https://questions.tripos.org/part-ii/2020-87/

**Comment by ArthurConmy** [comment]
The first parts are all bookwork. Then for the final part, basically we want to make all this rubbish into some language with axioms and apply compactness. This can probably be shortened, but the examiner's report suggests it's a pain regardless: 

Have arity 0 functions $f_1 , ..., f_{100}$ representing the symbols 1, ... , 100, as well as arity 0 functions $x$ for each $x \in X$, and finally arity 1 $F$

Have axioms (5 groups):
$f_i \neq f_j$ whenever $i \neq j$
$x \neq f_i$ for all $x\in X$ and all $i$
$x \neq x'$ for $x \neq x'$ both in $X$
$F(x) \neq F(\ell)$ for all $\ell \in L_x$ and $x \in X$
$( F(x) = f_1 ) \vee ... \vee (F(x) = f_{100})$ for all $x \in X$

Then finite subsets of these axioms will only mention finitely many $x \in X$, and so have this be our $Y$ in the notation of the assumption, and implement the arity 0 and arity 1 functions with unique elements for all $f_1 , ... ,f_{100}$, and arbitrary $x \not \in Y$.
